sdk/core/core/test/cmocka/test_az_http.c

```diff
@@ -357,6 +357,58 @@ enable_precondition_check_tests()
       az_http_request_append_header(&hrb, AZ_SPAN_FROM_STR(" \t\r"), AZ_SPAN_NULL));
 }
 
+static void test_http_request_header_validation(void** state)
+{
+  (void)state;
+  {
+    uint8_t header_buf[(2 * sizeof(az_pair))];
+    memset(header_buf, 0, sizeof(header_buf));
+
+    az_span url_span = AZ_SPAN_FROM_STR("some.url.com");
+    az_span header_span = AZ_SPAN_FROM_BUFFER(header_buf);
+    _az_http_request hrb;
+
+    TEST_EXPECT_SUCCESS(az_http_request_init(
+        &hrb,
+        &az_context_app,
+        az_http_method_get(),
+        url_span,
+        az_span_size(url_span),
+        header_span,
+        AZ_SPAN_FROM_STR("body")));
+
+    assert_precondition_checked(az_http_request_append_header(
+        &hrb, AZ_SPAN_FROM_STR("(headerName)"), hrb_header_content_type_token));
```

Comment:
nit: add other asserts on the contents of the `header_span`/`hrb`

Same below.

Reply:
done